### DISCUSSION TITLE
Cannot read bucket location. (S3 (Credentials from AWS Command Line Interface))

### DISCUSSION THREAD

**Discussion by eswolinsky3241** [iterate-ch/cyberduck#16155] [Q&A] [answered]
**Describe the bug**
We use AWS Identity Center to log in users of the AWS CLI. I am having an issue with Cyberduck version 9.0.0 where, after my SSO credentials expire and I log in again to refresh them, my connection attempts using the `S3 (Credentials from AWS Command Line Interface)` profile fail with the following error:

<img width="269" alt="image" src="https://github.com/user-attachments/assets/253e181f-e663-4dc8-ab08-9d157af709c8">

However, running the following CLI command returns data (I have confirmed that the profile I am using to run the CLI command matches the one I used to configure Cyberduck)
```
aws s3 ls s3://atx-archive-prod/ --profile Tech-Admins-252872867134
```

**To Reproduce**
1. Connect to S3 bucket using Cyberduck. Connection is successful.
2. Log out of SSO to simulate credentials expiring: `aws sso logout`.
3. Log back into SSO: `aws sso login`
4. Error is now raised when trying to connect to S3 with Cyberduck.
5. After about 10 minutes, Cyberduck is able to connect again.

**Expected behavior**
Cyberduck should use the currently cached credentials to connect to S3. 

**Screenshots**

**Desktop (please complete the following information):**
 - OS: macOS
 - Version: Sonoma

**Log Files**
Please include the [log file](https://docs.cyberduck.io/cyberduck/support/#logging-output). You can reach the logging output in _Preferences → Connection_.
[cyberduck.log](https://github.com/user-attachments/files/16240698/cyberduck.log)

**Additional context**

**Comment by eswolinsky3241** [comment]
> After about 10 minutes, Cyberduck is able to connect again.

After experimenting some more, it seems that it's not after some length of time that Cyberduck recognizes the new AWS keys. If I run any command in my terminal after logging in (i.e. `aws sts get-caller-identity`) Cyberduck is then able to connect successfully. Ideally this wouldn't be necessary to refresh my credentials in Cyberduck.

**Comment by dkocher** [comment, marked as answer]
This is according to the [documentation](https://docs.cyberduck.io/protocols/s3/#aws-iam-identity-center). We are aware of the mediocre usability and track this in #13377.
